### ISSUE TITLE
glmnet engine does not accept one row predictions

### ISSUE THREAD

**Issue by mrepetto94** [tidymodels/parsnip#407]
## The problem

I'm tryng to do one row prediction with `linear_reg()` models, with the default engine it works, but with `glmnet` do not.

## Reproducible example

``` r
library(tidyverse)
library(tidymodels)
#> -- Attaching packages -------------------------------------- tidymodels 0.1.2 --
#> v broom     0.7.3      v recipes   0.1.15
#> v dials     0.0.9      v rsample   0.0.8 
#> v infer     0.5.3      v tune      0.1.2 
#> v modeldata 0.1.0      v workflows 0.2.1 
#> v parsnip   0.1.4      v yardstick 0.0.7
#> -- Conflicts ----------------------------------------- tidymodels_conflicts() --
#> x scales::discard() masks purrr::discard()
#> x dplyr::filter()   masks stats::filter()
#> x recipes::fixed()  masks stringr::fixed()
#> x dplyr::lag()      masks stats::lag()
#> x yardstick::spec() masks readr::spec()
#> x recipes::step()   masks stats::step()
workflow() %>%
  add_recipe(recipe(mpg ~ ., data = mtcars)) %>%
  add_model(linear_reg(penalty = 0) %>% set_engine("glmnet")) %>%
  fit(mtcars) %>%
  predict(tail(mtcars, 1))
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'as.matrix': Cholmod error 'X and/or Y have wrong dimensions' at file ../MatrixOps/cholmod_sdmult.c, line 90
```

<sup>Created on 2021-01-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
<details>
<summary>
Session info
</summary>

``` r
sessionInfo()
#> R version 4.0.3 (2020-10-10)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 18363)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=Italian_Italy.1252  LC_CTYPE=Italian_Italy.1252   
#> [3] LC_MONETARY=Italian_Italy.1252 LC_NUMERIC=C                  
#> [5] LC_TIME=Italian_Italy.1252    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices datasets  utils     methods   base     
#> 
#> other attached packages:
#>  [1] yardstick_0.0.7  workflows_0.2.1  tune_0.1.2       rsample_0.0.8   
#>  [5] recipes_0.1.15   parsnip_0.1.4    modeldata_0.1.0  infer_0.5.3     
#>  [9] dials_0.0.9      scales_1.1.1     broom_0.7.3      tidymodels_0.1.2
#> [13] forcats_0.5.0    stringr_1.4.0    dplyr_1.0.2      purrr_0.3.4     
#> [17] readr_1.4.0      tidyr_1.1.2      tibble_3.0.4     ggplot2_3.3.3   
#> [21] tidyverse_1.3.0 
#> 
#> loaded via a namespace (and not attached):
#>  [1] fs_1.5.0           lubridate_1.7.9.2  DiceDesign_1.8-1   httr_1.4.2        
#>  [5] tools_4.0.3        backports_1.2.0    R6_2.5.0           rpart_4.1-15      
#>  [9] DBI_1.1.0          colorspace_2.0-0   nnet_7.3-14        withr_2.3.0       
#> [13] tidyselect_1.1.0   compiler_4.0.3     glmnet_4.0-2       cli_2.2.0         
#> [17] rvest_0.3.6        xml2_1.3.2         digest_0.6.27      rmarkdown_2.6     
#> [21] pkgconfig_2.0.3    htmltools_0.5.0    parallelly_1.22.0  lhs_1.1.1         
#> [25] dbplyr_2.0.0       highr_0.8          rlang_0.4.9        readxl_1.3.1      
#> [29] rstudioapi_0.13    shape_1.4.5        generics_0.1.0     jsonlite_1.7.2    
#> [33] magrittr_2.0.1     Matrix_1.2-18      Rcpp_1.0.5         munsell_0.5.0     
#> [37] fansi_0.4.1        GPfit_1.0-8        lifecycle_0.2.0    furrr_0.2.1       
#> [41] pROC_1.16.2        stringi_1.5.3      yaml_2.2.1         MASS_7.3-53       
#> [45] plyr_1.8.6         grid_4.0.3         parallel_4.0.3     listenv_0.8.0     
#> [49] crayon_1.3.4       lattice_0.20-41    haven_2.3.1        splines_4.0.3     
#> [53] hms_0.5.3          ps_1.5.0           knitr_1.30         pillar_1.4.7      
#> [57] codetools_0.2-16   reprex_0.3.0       glue_1.4.2         evaluate_0.14     
#> [61] renv_0.12.3        modelr_0.1.8       vctrs_0.3.6        foreach_1.5.1     
#> [65] cellranger_1.1.0   gtable_0.3.0       future_1.21.0      assertthat_0.2.1  
#> [69] xfun_0.19          gower_0.2.2        prodlim_2019.11.13 class_7.3-17      
#> [73] survival_3.2-7     timeDate_3043.102  iterators_1.0.13   hardhat_0.1.5     
#> [77] lava_1.6.8.1       globals_0.14.0     ellipsis_0.3.1     ipred_0.9-9
```

</details>

**Comment by juliasilge** [comment]
This bug 🐛  was fixed in #395. Can you install from GitHub and try again?

```r
devtools::install_github("tidymodels/parsnip")
```


``` r
library(tidymodels)
#> ── Attaching packages ───────────────────────────────── tidymodels 0.1.2.9000 ──
#> ✓ broom     0.7.3           ✓ recipes   0.1.15.9000
#> ✓ dials     0.0.9           ✓ rsample   0.0.8      
#> ✓ dplyr     1.0.2           ✓ tibble    3.0.4      
#> ✓ ggplot2   3.3.3           ✓ tidyr     1.1.2      
#> ✓ infer     0.5.3           ✓ tune      0.1.2.9000 
#> ✓ modeldata 0.1.0           ✓ workflows 0.2.1      
#> ✓ parsnip   0.1.4.9000      ✓ yardstick 0.0.7      
#> ✓ purrr     0.3.4
#> ── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()

workflow() %>%
  add_recipe(recipe(mpg ~ ., data = mtcars)) %>%
  add_model(linear_reg(penalty = 0) %>% set_engine("glmnet")) %>%
  fit(mtcars) %>%
  predict(tail(mtcars, 1))
#> # A tibble: 1 x 1
#>   .pred
#>   <dbl>
#> 1  24.4
```

<sup>Created on 2021-01-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mrepetto94** [comment]
Indeed the latest version from Github fixed the bug 👍
